extras/kms/kms.go

```diff
@@ -40,15 +40,20 @@ type Kms struct {
 
 	externalWrapperCache sync.Map
 
-	purposes []KeyPurpose
-	repo     *repository
+	purposes  []KeyPurpose
+	repo      *repository
+	withCache bool
 }
 
 // New takes in a reader, writer and a list of key purposes it will support.
 // Every kms will support a KeyPurposeRootKey by default and it doesn't need to
-// be passed in as one of the supported purposes.  No options are currently
-// supported.
-func New(r dbw.Reader, w dbw.Writer, purposes []KeyPurpose, _ ...Option) (*Kms, error) {
+// be passed in as one of the supported purposes.
+//
+// The WithCache(bool) option is supported and if enabled then a cache of the
+// wrappers is enabled.  Once enabled, the cache will speed up calls to
+// GetWrapper(...) but the caller also becomes responsible to reset/clear the
```

Comment:
how does a scope get deleted from the database?

Reply:
that's outside the "scope" of this package.